File: tests_C++/test2.cpp
/**
 * Nullity test of Gradient operator
 */

#include "mole.h"
#include <iostream>

void run_nullity_test(int k, Real tol) {
  int m = 2 * k + 1;
  Real dx = 1;

  Gradient G(k, m, dx);
  vec field(m + 2, fill::ones);

  vec sol = G * field;

  if (norm(sol) > tol) {
    cout << "\033[1;31mTest FAILED!\033[0m\n";
    exit(1);
  }
}

int main() {
  Real tol = 1e-12;

  for (int k : {2, 4, 6})
    run_nullity_test(k, tol);

  cout << "\033[1;32mTest PASSED!\033[0m\n";

  return 0;
}

File: tests_MATLAB/test2.m
% Nullity test of Gradient operator

addpath('../mole_MATLAB')

k = 2;
m = 2 * k + 1;
dx = 1;
tol = 1e-12;

G = grad(k, m, dx);

field = ones(m + 2, 1);

sol = G * field;

if (norm(sol) < tol)
    fprintf("Test PASSED!\n");
else
    fprintf("Test FAILED!");
end